runtimes/eden/src/migrations.rs

```diff
@@ -100,27 +143,63 @@ impl OnRuntimeUpgrade for MoveValidatorsSetToInvulnerables {
 		{
-			Err("Failed to remove ValidatorsSet and/or Poa")
-		} else {
-			let invulnerables = pallet_collator_selection::pallet::Pallet::<Runtime>::invulnerables();
-			log::info!(
-				target: TRY_RUNTIME,
-				"CollatorSelection::Invulnerables are {:?}",
-				invulnerables
-			);
-			let candidacy_bond = pallet_collator_selection::pallet::Pallet::<Runtime>::candidacy_bond();
-			log::info!(
-				target: TRY_RUNTIME,
-				"CollatorSelection::candidacy_bond is {:?}",
-				candidacy_bond
-			);
-			let validators: BoundedVec<AccountId, MaxMembers> = Decode::decode(&mut state.as_slice()).unwrap();
-			if invulnerables == validators && candidacy_bond == CANDIDACY_BOND {
-				log::info!(target: TRY_RUNTIME, "MoveValidatorsSetToInvulnerables was successful");
-				Ok(())
-			} else {
-				Err("CollatorSelection::Invulnerables are not the same as ValidatorsSet::Members")
+			return Err("Failed to remove ValidatorsSet and/or Poa");
+		}
+
+		let invulnerables = pallet_collator_selection::pallet::Pallet::<Runtime>::invulnerables();
+		log::info!(
+			target: TRY_RUNTIME,
+			"CollatorSelection::Invulnerables are {:?}",
+			invulnerables
+		);
+		let candidacy_bond = pallet_collator_selection::pallet::Pallet::<Runtime>::candidacy_bond();
+		log::info!(
+			target: TRY_RUNTIME,
+			"CollatorSelection::candidacy_bond is {:?}",
+			candidacy_bond
+		);
+		let validators: BoundedVec<AccountId, MaxMembers> = Decode::decode(&mut state.as_slice()).unwrap();
+		if invulnerables != validators || candidacy_bond != CANDIDACY_BOND {
+			return Err("CollatorSelection::Invulnerables are not the same as ValidatorsSet::Members");
+		}
+		log::info!(target: TRY_RUNTIME, "MoveValidatorsSetToInvulnerables was successful");
+
+		if <cumulus_pallet_dmp_queue::pallet::Pallet<Runtime>>::current_storage_version()
+			!= <cumulus_pallet_dmp_queue::pallet::Pallet<Runtime>>::on_chain_storage_version()
+		{
+			log::info!(target: TRY_RUNTIME, "DMP Queue storage version is not updated");
+			return Err("DMP Queue storage version is not updated");
+		}
+		log::info!(target: TRY_RUNTIME, "DMP Queue storage version was good");
+
+		if <cumulus_pallet_xcmp_queue::pallet::Pallet<Runtime>>::current_storage_version()
+			!= <cumulus_pallet_xcmp_queue::pallet::Pallet<Runtime>>::on_chain_storage_version()
+		{
+			log::info!(target: TRY_RUNTIME, "XCMP Queue storage version is not updated");
+			return Err("XCMP Queue storage version is not updated");
+		}
+		log::info!(target: TRY_RUNTIME, "XCMP Queue storage version was good");
+
+		if <pallet_contracts::pallet::Pallet<Runtime>>::current_storage_version()
+			!= <pallet_contracts::pallet::Pallet<Runtime>>::on_chain_storage_version()
+		{
+			log::info!(target: TRY_RUNTIME, "Contracts storage version is not updated");
+			return Err("Contracts storage version is not updated");
+		}
+		log::info!(target: TRY_RUNTIME, "Contracts storage version was good");
+
+		if let Some(xcm_version) =
```

Comment:
Use this a a final statement and embed the Ok(()) in the correct place and end the others without return, then the compiler will check all arms are covered for us.

Reply:
I think because I didn't have `else` statements in my structure, I had to `return` explicitly.